core/src/main/java/software/amazon/awssdk/http/pipeline/stages/MakeAsyncHttpRequestStage.java

```diff
@@ -167,19 +182,25 @@ public void onStream(Publisher<ByteBuffer> publisher) {
 
         @Override
         public void exceptionOccurred(Throwable throwable) {
-            responseHandler.exceptionOccurred(throwable);
+            runAndLogError("SdkHttpResponseHandler threw an exception.",
```

Comment:
We weren't notifying the future in certain cases when the response handlers threw exceptions.